File: web/server/restapihandlers/volume.py
# Copyright 2021 Peppy Player peppy.player@gmail.com
# 
# This file is part of Peppy Player.
# 
# Peppy Player is free software: you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation, either version 3 of the License, or
# (at your option) any later version.
# 
# Peppy Player is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
# 
# You should have received a copy of the GNU General Public License
# along with Peppy Player. If not, see <http://www.gnu.org/licenses/>.

import json

from tornado.web import RequestHandler
from util.config import PLAYER_SETTINGS, VOLUME
from ui.state import State

class VolumeHandler(RequestHandler):
    def initialize(self, peppy):
        self.config = peppy.util.config
        self.peppy = peppy

    def get(self):
        try:
            v = {VOLUME: self.config[PLAYER_SETTINGS][VOLUME]}
            self.write(json.dumps(v))
        except:
            self.set_status(500)
            return self.finish()

    def put(self):
        try:
            d = json.loads(self.request.body)
            state = State()
            v = d[VOLUME]
            state.position = v
            self.peppy.set_volume(state)
            self.config[PLAYER_SETTINGS][VOLUME] = v
            screen = self.peppy.screens[self.peppy.current_player_screen]
            if screen:
                screen.refresh_volume()
                screen.redraw_observer()
        except:
            self.set_status(500)
            return self.finish()

File: web/client/playerdev/src/min/Volume.js
/* Copyright 2024 Peppy Player peppy.player@gmail.com
 
This file is part of Peppy Player.
 
Peppy Player is free software: you can redistribute it and/or modify
it under the terms of the GNU General Public License as published by
the Free Software Foundation, either version 3 of the License, or
(at your option) any later version.
 
Peppy Player is distributed in the hope that it will be useful,
but WITHOUT ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
GNU General Public License for more details.
 
You should have received a copy of the GNU General Public License
along with Peppy Player. If not, see <http://www.gnu.org/licenses/>.
*/

import React from "react";
import Drawer from '@mui/material/Drawer';
import Slider from '@mui/material/Slider';
import IconButton from '@mui/material/IconButton';
import { VolumeSliderStyle, VolumePanel, Mute } from './mincss';
import { setVolume, getIconUrl, getIconSelectedUrl } from "../Rest";

export default class Volume extends React.Component {

  render() {
    const { openVolume, toggleVolume, volume, setPosition, mute, setMute } = this.props;

    return (
      <Drawer open={openVolume} onClose={toggleVolume(false)}
        anchor='bottom'
        PaperProps={{
          style: {
            height: '86px',
            display: 'flex',
            flexDirection: 'row',
            alignItems: 'center',
            justifyContent: 'center'
          }
        }}
      >
        <div style={VolumePanel}>
          <Slider
            disabled={mute}
            orientation="horizontal"
            aria-label="Volume"
            valueLabelDisplay="auto"
            value={volume}
            defaultValue={0}
            size='small'
            sx={VolumeSliderStyle}
            onChange={setPosition}
            onChangeCommitted={setVolume}
          />
          <IconButton aria-label="mute" style={{ marginLeft: '0.5rem' }} onClick={setMute}>
            <img src={mute ? getIconSelectedUrl('volume-mute') : getIconUrl('volume-mute')} key="mute" alt="mute" style={Mute} />
          </IconButton>
        </div>
      </Drawer>
    );
  }
}
